hopsworks-api/src/main/java/io/hops/hopsworks/api/user/UsersResource.java

```diff
@@ -192,6 +200,42 @@ public Response changeLoginCredentials(
     return Response.ok().entity(json).build();
   }
 
+  @POST
+  @Path("apiKey")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Stores a third-party API key for user")
+  public Response addApiKey(@FormParam("name") String keyName, @FormParam("key") String key,
+      @Context SecurityContext sc) throws UserException {
+    Users user = jWTHelper.getUserPrincipal(sc);
+    thirdPartyApiKeysController.addApiKey(user, keyName, key);
+    
+    RESTApiJsonResponse response = new RESTApiJsonResponse();
+    response.setSuccessMessage("Added new API key");
+    return Response.ok().entity(response).build();
+  }
+  
+  @GET
+  @Path("apiKey")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Retrieves all third-party API key names of a user", response = ThirdPartyApiKeyDTO.class)
+  public Response getAllApiKeys(@Context SecurityContext sc) throws UserException {
+    Users user = jWTHelper.getUserPrincipal(sc);
+    List<ThirdPartyApiKeyPlaintext> apiKeys = thirdPartyApiKeysController.getAllApiKeysForUser(user);
+    ThirdPartyApiKeyDTO dto = thirdPartyApiKeysBuilder.build(apiKeys);
+    return Response.ok().entity(dto).build();
+  }
+  
+  @DELETE
+  @Path("apiKey/{keyName}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Deletes an API key by its name")
+  public Response deleteApiKey(@PathParam("keyName") String keyName, @Context SecurityContext sc)
```

Comment:
@kouzant how about having an endpoint that deletes all the apiKeys?

Reply:
I could add one to delete all API keys for user